app/client/src/tests/unit/App.test.js

```diff
@@ -0,0 +1,153 @@
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+import '../index.css';
+import App from '../../App';
+import {BrowserRouter} from 'react-router-dom';
+import UserEvent from '@testing-library/user-event';
+import { unmountComponentAtNode } from 'react-dom';
+
+/*
+  Setup of DOM container
+*/
+
+let container = null;
+
+function containerInit(){
+  container = document.createElement("div");
+  document.body.appendChild(container);
+
+  render( <React.StrictMode>
+    <BrowserRouter>
+    <App/>
+    </BrowserRouter>
+    </React.StrictMode>, container
+  );
+}
+
+function containerRemove(){
+    unmountComponentAtNode(container);
+    container.remove();
+    container = null;
+}
+
+
+
+describe('Testing log in, sign up, register front-end behaviours', () => {
+  
+  beforeEach(() => {
+    containerInit();
+  });
+
+  afterEach(() => {
+    containerRemove();
+  });
+
+  test('sign up page loads with all four fields and four labels', () => {
+    expect(screen.getByText('IDIR:')).toBeInTheDocument();
+    expect(screen.getByPlaceholderText('ID')).toBeInTheDocument();
+    expect(screen.getByText('Email:')).toBeInTheDocument();
+    expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
+    expect(screen.getByText('Password:')).toBeInTheDocument();
+    expect(screen.getByText('Re-Enter Password:')).toBeInTheDocument();
+    expect(screen.queryAllByPlaceholderText('Password')).toHaveLength(2); //one per password field
+  });
+
+  test('entering bad passwords should open warning window', () => {
+    const passwordField = document.getElementById('psw');
```

Comment:
do we need to add a TODO here on the number of possible password iterations we can use to break the field based on the constraints provided?

Reply:
Perhaps. I think we need to discuss this with the capstones to see if the password screening will be done client or server side in the end.

Reply:
@dbarkowsky It is currently only done client side but its in the backlog for this sprint to have it server-side as well.